### ISSUE TITLE
Add PLCs and sensors 

### ISSUE THREAD

**Issue by Domi1993** [Electrical-Age/ElectricalAge#334]
A PLC will be cool and sensors and different types of switches (Emergency stop, rotating switches...)
And some thing to make an elevator or something other like a crane.. 

I'm missing the auxiliary contacts to attach on relays to get a feedback of their state, combined with Signal relays/Contactors. To made some custom controls with it (elevatorcontrol)

**Comment by Baughn** [comment]
PLCs are coming. I have a good deal of preliminary work to do first, so they aren't coming _soon_, but they're coming.

I'm uncertain what you mean by relay feedback, though. There isn't a failure chance... Ah, because of the hysteresis?

**Comment by Domi1993** [comment]
No i mean that i can put a signal wire to the relay for an signal contact that do the same thing like the big contact so i can activate a light or something to show that the relay is closed

**Comment by Domi1993** [comment]
Just look for auxiliary contacs in wikipedia or in google

**Comment by Domi1993** [comment]
Look here https://de.m.wikipedia.org/wiki/Datei:Selbsthaltung.gif

This is a typical usage for auxiliary contacts from a relay, the contact K1 will be used to make a rs flip flop function for the relay

**Comment by cm0x4D** [comment]
3 of the 4 possible contact directions of the current relay are used. How should we add another two? This would mean we have to create a 2-block relay. If you need flip-flop functionality, you can just take two relays in order to implement it. Another possibility would be to add self-holding functionality to the relay as an additional mode to normal and inverse mode...

**Comment by Domi1993** [comment]
Yes but there are no Signalrelays i mean when there are signalrelays my problem is solved.

The problem is that normal relays cannot switch signalcables. How do you think i should made a RS-flip-flop with relays that not accept signalwires at them Maincontact? I dont  want to use everytime a signalprocessor.

**Comment by cm0x4D** [comment]
I added a signal relay to the branch "features/signalRelay". The model is just a flat version of the normal relay. What is actually still missing is a signal source in order to drive the self-holding mechanism, I used a switch (Always on) to power the relay contact. Not sure if the feature will make it in the main branch, have to discuss this with other mod devs...
Feel free to test the new relay (Signal Relay).

**Comment by Domi1993** [comment]
How i can get to this branch to get the relay

**Comment by cm0x4D** [comment]
You need to have installed git and gradle.

Then just type into a terminal:

```
git clone https://github.com/Dolu1990/ElectricalAge.git
cd ElectricalAge
git checkout features/signalRelay
gradle runClient
```

**Comment by Baughn** [comment]
Pop by IRC if you have trouble with that. Well, and have patience; the channel isn't very active yet, and I'm flying today. So maybe tomorrow. :D

**Comment by Domi1993** [comment]
I tried the git and gradle in GitBash console gradle runClient says me unknow command must i enter in another console gradle runClient? In which console?

**Comment by cm0x4D** [comment]
Ok, seems you are on Windows. Can't help there, maybe ask on IRC.

**Comment by bloxgate** [comment]
Try ./gradlew runClient in the eln directory

**Comment by Baughn** [comment]
"./gradlew" only works on Unix-like systems. I think there's a .bat wrapper for Windows, though..?

**Comment by Domi1993** [comment]
??? I don't know anything about computer related things like programming or .bat wrappers

**Comment by Domi1993** [comment]
But you can ask me things about Eletricity PLCs and Relays and the other Electromechanics.

But Computer And Electronics i'm know only basics

**Comment by Domi1993** [comment]
Can you say me why Minecraft is only using 1% GPU Power of my Intel graphics 4400?

It's lagging but a few months ago it runned very good, i don't know what i can do for this problem it is on Win 10 when it worked good it was on win 10 too

**Comment by bloxgate** [comment]
./gradlew has been working fine for me on my Windows 10 system.

**Comment by Domi1993** [comment]
Just one little question have you programmed the batteries according to a real battery characteristic when yes for which battery type. I think you have do something custom for that, and nothing like a Pb battery with it's real voltage characteristics

**Comment by Domi1993** [comment]
The Signal relay is awesome it's needed in an Official ELN-Release

**Comment by jrddunbr** [comment]
This looks completed? Minus a elevator and some buttons?
